src/handler-context/Config/index.ts

```diff
@@ -0,0 +1,40 @@
+import { Json } from "../../shared/types";
+import { ConfigFileNames, ConfigClient, PickedFiles } from "./types";
+
+export class Config<TFileName extends ConfigFileNames> {
+  private readonly client: ConfigClient;
+  private readonly files: ConfigFileNames[];
+  private readonly promises: Array<Promise<[ConfigFileNames, Json]>>;
+  private cache: PickedFiles<TFileName> | undefined;
+
+  constructor(client: ConfigClient, files: TFileName[]) {
+    this.client = client;
+    this.files = files;
+    this.promises = this.spawnPromises();
+  }
+
+  private readonly spawnPromises = (): Array<
+    Promise<[ConfigFileNames, Json]>
+  > => {
+    if (process.env.CONFIG_BUCKET === undefined)
```

Comment:
This should be something the ConfigClient should care about. This module would ideally not care about where the config comes from.

Reply:
agreed, I've moved it